src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -261,17 +234,84 @@ func (h *PublicApiHandler) DetachScalingPolicy(w http.ResponseWriter, r *http.Re
+	serviceInstance, err := h.bindingdb.GetServiceInstanceByAppId(appId)
+	if err != nil {
+		logger.Error("Failed to find service instance for app", err)
+		writeErrorResponse(w, http.StatusInternalServerError, "Error retrieving service instance")
+		return errors.New("error retrieving service instance")
+	}
+
+	if serviceInstance != nil && serviceInstance.DefaultPolicy != "" {
+		return h.saveDefaultPolicy(w, r, logger, appId, serviceInstance)
+	}
+
+	return nil
+}
+
+func (h *PublicApiHandler) saveDefaultPolicy(w http.ResponseWriter, r *http.Request, logger lager.Logger, appId string, serviceInstance *models.ServiceInstance) error {
+	policyStr := serviceInstance.DefaultPolicy
+	policyGuidStr := serviceInstance.DefaultPolicyGuid
+	logger.Info("saving default policy json for app", lager.Data{"policy": policyStr})
+
+	var policy *models.ScalingPolicy
+	if err := json.Unmarshal([]byte(policyStr), &policy); err != nil {
+		h.logger.Error("default policy invalid", err, lager.Data{"appId": appId, "policy": policyStr})
+		writeErrorResponse(w, http.StatusInternalServerError, "Default policy not valid")
+		return errors.New("default policy not valid")
+	}
+
+	if err := h.policydb.SaveAppPolicy(r.Context(), appId, policy, policyGuidStr); err != nil {
+		logger.Error("failed to save policy", err, lager.Data{"policy": policyStr})
+		writeErrorResponse(w, http.StatusInternalServerError, "Error attaching the default policy")
+		return errors.New("error attaching the default policy")
+	}
+
+	//while there is synchronization between policy and schedule, so creating schedule error does not break
+	//the whole creating binding process
+	logger.Info("creating/updating schedules", lager.Data{"policy": policyStr})
+	if err := h.schedulerUtil.CreateOrUpdateSchedule(r.Context(), appId, policy, policyGuidStr); err != nil {
+		logger.Error("failed to create/update schedules", err, lager.Data{"policy": policyStr})
+		writeErrorResponse(w, http.StatusInternalServerError, fmt.Sprintf("Failed to update schedule:%s", err))
+		return errors.New("failed to update schedule")
+	}
+
+	return nil
+}
+
+func (h *PublicApiHandler) proxyRequest(logger lager.Logger, appId string, metricType string, w http.ResponseWriter, req *http.Request, parameters *url.Values, requestDescription string) {
+	reqUrl := req.URL
+	r := routes.NewRouter()
+	router := r.CreateEventGeneratorRoutes()
+	if router == nil {
+		panic("Failed to create event generator routes")
+	}
+
+	route := router.Get(routes.GetAggregatedMetricHistoriesRouteName)
+	path, err := route.URLPath("appid", appId, "metrictype", metricType)
+	if err != nil {
+		logger.Error("Failed to create path", err)
+		panic(err)
+	}
+
+	aUrl := h.conf.EventGenerator.EventGeneratorUrl + path.RequestURI() + "?" + parameters.Encode()
+	req, _ = http.NewRequest("GET", aUrl, nil)
+
+	if h.conf.CfInstanceCert != "" {
+		cert := auth.NewCert(h.conf.CfInstanceCert)
+		req.Header.Set("X-Forwarded-Client-Cert", cert.GetXFCCHeader())
+	}
```

Comment:
We should not set ourselves the XFCC header!

We should set up mTLS, see https://github.com/cloudfoundry/app-autoscaler-release/blob/44797b36264693c9b16d3fbc464362f8118cc421/src/acceptance/assets/app/go_app/internal/app/custom_metrics.go#L164-L192 and let gorouter or whoever terminates TLS set the header.

Reply:
fix in 39d1f3e9f